# Goal Split Subgroup: Transfer Resolution Model

Consequent Reversibility was not included in the final escape models because, in this subgroup structure, it is partially encoded by Goal_Type and produced unstable estimates when entered together with Goal_Type.

Import Libraries

In [124]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Goal Split Subgroup

1. Create resolved_transfer (goals and no_goals)
2. Counts and Proportions (goals and no_goals)
3. Logistic Models

    - Goals

        Main Effect Models:

        - RT01: Focus
        - RT02: Agent
        - RT03: Goal_Type

        Additive Models:

        - RT04: Focus + Agent
        - RT05: Focus + Goal_Type
        - RT06: Agent + Goal_Type

        Interactive Models:

        - RT07: Focus * Agent
        - RT08: Focus * Goal_Type
        - RT09: Agent * Goal_Type
        - RT10: Focus * Agent + Goal_Type

    - No_goals

        Main Effect Models:

        - RT11:
        - RT12:

        Additive Model:

        - RT13:

        Interactive Model:

        - RT14:

4. Model Comparisons
5. Summary Table

Read Data

In [125]:
subgroup_theoretical = pd.read_csv("../../data/processed/subgroup_theoretical.csv")

1. Create resolved_transfer

Binary Outcome: transfer responses from those observations that managed to escape L2_other options.

In [126]:
subgroup_theoretical["escape_L2"] = (
    subgroup_theoretical["Response_Full"] != "L2_other"
).astype(int)

In [127]:
escapees = subgroup_theoretical[subgroup_theoretical["escape_L2"] == 1].copy()

In [128]:
escapees["resolved_transfer"] = (escapees["Response_Full"] == "L1_transfer").astype(int)

Split escapees data into goals vs. no_goals

In [129]:
goals_escapees = escapees[escapees["Goal_Type"] != "no_goal"].copy()

In [130]:
no_goal_escapees = escapees[escapees["Goal_Type"] == "no_goal"].copy()

2. Counts and Proportions for goals and no_goals

General resolved_correct reponses count:

In [131]:
resolved_transfer_counts = pd.crosstab(
    escapees["resolved_transfer"],
    escapees["Response_Full"],
    margins=True)

resolved_transfer_counts

Response_Full,L1_transfer,correct,missing_response,All
resolved_transfer,,,,
0,0,61,3,64
1,96,0,0,96
All,96,61,3,160


resolved_transfer responses count for goals:

In [132]:
goals_resolved_transfer_counts = pd.crosstab(
    goals_escapees["resolved_transfer"],
    goals_escapees["Response_Full"],
    margins=True)

goals_resolved_transfer_counts

Response_Full,L1_transfer,correct,missing_response,All
resolved_transfer,,,,
0,0,43,1,44
1,62,0,0,62
All,62,43,1,106


resolved_transfer responses count for no_goals:

In [133]:
no_goal_resolved_transfer_counts = pd.crosstab(
    no_goal_escapees["resolved_transfer"],
    no_goal_escapees["Response_Full"],
    margins=True)

no_goal_resolved_transfer_counts

Response_Full,L1_transfer,correct,missing_response,All
resolved_transfer,,,,
0,0,18,2,20
1,34,0,0,34
All,34,18,2,54


General resolved_transfer responses proportions:

In [134]:
resolved_transfer_props = escapees["resolved_transfer"].value_counts(normalize = True)
resolved_transfer_props

resolved_transfer
1    0.6
0    0.4
Name: proportion, dtype: float64

From 160 observations, 40% of responses that had escaped *L2_other* options were resolved by means of transfer. 

resolved_correct responses proportions for goals:

In [135]:
goals_resolved_transfer_props = goals_escapees["resolved_transfer"].value_counts(normalize=True)
goals_resolved_transfer_props

resolved_transfer
1    0.584906
0    0.415094
Name: proportion, dtype: float64

From 106 observations, 41.5% of goal responses that had escaped *L2_other* options were resolved by means of transfer. Slightly higher than resolving correctly (40.5%) but not that much. 

resolved_correct responses proportions for no_goals:

In [136]:
no_goal_resolved_transfer_props = no_goal_escapees["resolved_transfer"].value_counts(normalize=True)
no_goal_resolved_transfer_props

resolved_transfer
1    0.62963
0    0.37037
Name: proportion, dtype: float64

From 54 observations, 37% of no_goal responses that had escaped *L2_other* options were resolved by means of transfer. Also higher than resolving correctly (33.3%) but not that much either.

- The proportion distribution of transfer resolution responses is higher when there is a goal oriented representation (41.5%) as opposed to when there isn't a representation of a goal (37%). 
- However, the gap narrows compared to the distribution of correct resolution responses between goal oriented vs. no goal representations.

Counts and Proportions by Condition:

In [137]:
counts_condition = pd.crosstab(
    [escapees["Goal_Type"], escapees["Agent"], escapees["Focus"]],
    escapees["resolved_transfer"],
    margins = True)

counts_condition

resolved_transfer               0   1  All
Goal_Type         Agent Focus             
goal_frequent     0     I      10   6   16
                        They    5   7   12
                  1     I       7   7   14
                        They    3   5    8
goal_non_frequent 0     I       9   7   16
                        They    3  14   17
                  1     I       4  11   15
                        They    3   5    8
no_goal           0     I       8  13   21
                        They    3  10   13
                  1     I       5   9   14
                        They    4   2    6
All                            64  96  160

In [138]:
goals_counts_condition = pd.crosstab(
    [goals_escapees["Goal_Type"], goals_escapees["Agent"], goals_escapees["Focus"]],
    goals_escapees["resolved_transfer"],
    margins = True)

goals_counts_condition

resolved_transfer               0   1  All
Goal_Type         Agent Focus             
goal_frequent     0     I      10   6   16
                        They    5   7   12
                  1     I       7   7   14
                        They    3   5    8
goal_non_frequent 0     I       9   7   16
                        They    3  14   17
                  1     I       4  11   15
                        They    3   5    8
All                            44  62  106

In [139]:
no_goal_counts_condition = pd.crosstab(
    [no_goal_escapees["Goal_Type"], no_goal_escapees["Agent"], no_goal_escapees["Focus"]],
    no_goal_escapees["resolved_transfer"],
    margins = True)

no_goal_counts_condition

resolved_transfer       0   1  All
Goal_Type Agent Focus             
no_goal   0     I       8  13   21
                They    3  10   13
          1     I       5   9   14
                They    4   2    6
All                    20  34   54

In [140]:
goals_props_condition = pd.crosstab(
    [goals_escapees["Goal_Type"], goals_escapees["Agent"]],
    goals_escapees["resolved_transfer"],
    normalize= "index")

goals_props_condition

resolved_transfer               0         1
Goal_Type         Agent                    
goal_frequent     0      0.535714  0.464286
                  1      0.454545  0.545455
goal_non_frequent 0      0.363636  0.636364
                  1      0.304348  0.695652

In [141]:
no_goal_props_condition = pd.crosstab(
    [no_goal_escapees["Goal_Type"], no_goal_escapees["Agent"]],
    no_goal_escapees["resolved_transfer"],
    normalize= "index")

no_goal_props_condition

resolved_transfer         0         1
Goal_Type Agent                      
no_goal   0        0.323529  0.676471
          1        0.450000  0.550000

COMMENT GENERAL PROPORTIONS

3. Logistic Models

Import Libraries

In [142]:
import statsmodels.formula.api as smf
from scipy.stats import chi2

Sanity checks:

In [143]:
escapees["resolved_transfer"].value_counts()

resolved_transfer
1    96
0    64
Name: count, dtype: int64

In [144]:
goals_escapees["resolved_transfer"].value_counts()

resolved_transfer
1    62
0    44
Name: count, dtype: int64

In [145]:
no_goal_escapees["resolved_transfer"].value_counts()

resolved_transfer
1    34
0    20
Name: count, dtype: int64

**Goals**

**Main Effect Models**

**RT01**

resolved_transfer ~ Focus

In [146]:
RT01 = smf.logit(
    "resolved_transfer ~ Focus",
    data=goals_escapees
    ).fit()

print(RT01.summary())

Optimization terminated successfully.
         Current function value: 0.662011
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      104
Method:                           MLE   Df Model:                            1
Date:                Thu, 23 Jul 2026   Pseudo R-squ.:                 0.02453
Time:                        18:26:41   Log-Likelihood:                -70.173
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                   0.06029
                    coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept         0.0328      0.256      0.128      0.898      -0.469       0.535
Focus[T.They]     0.

With goal split dataset, the effect of Focus on its own seems to be more defined almost reaching significant.

**RT02**

resolved_transfer ~ Agent

In [147]:
RT02 = smf.logit(
    "resolved_transfer ~ Agent",
    data=goals_escapees
    ).fit()

print(RT02.summary())

Optimization terminated successfully.
         Current function value: 0.676537
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      104
Method:                           MLE   Df Model:                            1
Date:                Thu, 23 Jul 2026   Pseudo R-squ.:                0.003127
Time:                        18:26:41   Log-Likelihood:                -71.713
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                    0.5024
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.2305      0.258      0.894      0.371      -0.275       0.736
Agent          0.2685      0.

Agency still can't do anything on its own; however, directing the analysis to goal data has improved it a lot compared to combined data (RT02 LLR p_value = 0.9999).

**RT03**

resolved_transfer ~ Goal_Type

In [148]:
RT03 = smf.logit(
    "resolved_transfer ~ Goal_Type",
    data=goals_escapees
    ).fit()

print(RT03.summary())

Optimization terminated successfully.
         Current function value: 0.665366
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      104
Method:                           MLE   Df Model:                            1
Date:                Thu, 23 Jul 2026   Pseudo R-squ.:                 0.01959
Time:                        18:26:41   Log-Likelihood:                -70.529
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                   0.09320
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                       6.516e-17      0.283    2.3e-16     

By eliminating no_goal, the model with Goal Type alone has improved a a lot. Goal_non_frequent keeps the same coefficient as with combined data (0.666) and the same p-value = 0.095.

**Additive Models**

**RT04**

resolved_transfer ~ Focus + Agent

In [149]:
RT04 = smf.logit(
    "resolved_transfer ~ Focus + Agent",
    data=goals_escapees
    ).fit()

print(RT04.summary())

Optimization terminated successfully.
         Current function value: 0.658047
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      103
Method:                           MLE   Df Model:                            2
Date:                Thu, 23 Jul 2026   Pseudo R-squ.:                 0.03037
Time:                        18:26:41   Log-Likelihood:                -69.753
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                    0.1125
                    coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept        -0.1460      0.323     -0.452      0.651      -0.779       0.487
Focus[T.They]     0.

Just as with combined data, the additive model Focus-Agent cannot quite make it. Focus keeps p-value improves a little compared to the model with Focus on its own. The model has improved a lot compared to the same with model with combined data. 

**RT05**

resolved_transfer ~ Focus + Goal_Type

In [150]:
RT05 = smf.logit(
    "resolved_transfer ~ Focus + Goal_Type",
    data=goals_escapees
    ).fit()

print(RT05.summary())

Optimization terminated successfully.
         Current function value: 0.649658
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      103
Method:                           MLE   Df Model:                            2
Date:                Thu, 23 Jul 2026   Pseudo R-squ.:                 0.04273
Time:                        18:26:41   Log-Likelihood:                -68.864
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                   0.04623
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                         -0.2983      0.331     -0.901     

Even though the coefficients do not make it to significance, the model is significant overall. Goal_non_frequent's coefficient and p-values are almost the same as the model with combined data; it is Focus what that has improved once the no_goal variation was removed. This is interesting as it does give way to consider that, despite not having enough power, Goal Type could after all play a role. 

**RT06**

resolved_transfer ~ Agent + Goal_Type

In [151]:
RT06 = smf.logit(
    "resolved_transfer ~ Agent + Goal_Type",
    data=goals_escapees
    ).fit()

print(RT06.summary())

Optimization terminated successfully.
         Current function value: 0.662848
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      103
Method:                           MLE   Df Model:                            2
Date:                Thu, 23 Jul 2026   Pseudo R-squ.:                 0.02330
Time:                        18:26:41   Log-Likelihood:                -70.262
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                    0.1871
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                         -0.1305      0.335     -0.389     

This model also improves compared to the combined dataset's. And, although goal_non_frequent's p-value doesn't reach significance (but it almost matches the combined data's at 0.095), it does show how it stands compared to Agency when it comes to resolving with transfer. 

**Interactive Models**

**RT07**

resolved_transfer ~ Focus * Agent

In [152]:
RT07 = smf.logit(
    "resolved_transfer ~ Focus * Agent",
    data=goals_escapees
    ).fit()

print(RT07.summary())

Optimization terminated successfully.
         Current function value: 0.646500
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      102
Method:                           MLE   Df Model:                            3
Date:                Thu, 23 Jul 2026   Pseudo R-squ.:                 0.04739
Time:                        18:26:41   Log-Likelihood:                -68.529
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                   0.07794
                          coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------------
Intercept              -0.3795      0.360     -1.054      0.292      -1.085       0.326
Fo

Compaired to the combined dataset, Focus p-value improves a little bit and so does the Agent's. The interaction's p-values increases this time while keeping the same coefficient, with the same magnitude and the same direction. This could just be as a result of reducing observations and not having enough power. The model's overall LLR p-value also improves compared to combined data although very slightly. 

**RT08**

resolved_transfer ~ Focus * Goal_Type

In [153]:
RT08 = smf.logit(
    "resolved_transfer ~ Focus * Goal_Type",
    data=goals_escapees
    ).fit()

print(RT08.summary())

Optimization terminated successfully.
         Current function value: 0.649498
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      102
Method:                           MLE   Df Model:                            3
Date:                Thu, 23 Jul 2026   Pseudo R-squ.:                 0.04297
Time:                        18:26:41   Log-Likelihood:                -68.847
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                    0.1031
                                                   coef    std err          z      P>|z|      [0.025      0.975]
----------------------------------------------------------------------------------------------------------------
Intercept                               

**RT09**

resolved_transfer ~ Agent * Goal_Type

In [154]:
RT09 = smf.logit(
    "resolved_transfer ~ Agent * Goal_Type",
    data=goals_escapees
    ).fit()

print(RT09.summary())

Optimization terminated successfully.
         Current function value: 0.662824
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      102
Method:                           MLE   Df Model:                            3
Date:                Thu, 23 Jul 2026   Pseudo R-squ.:                 0.02333
Time:                        18:26:41   Log-Likelihood:                -70.259
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                    0.3398
                                           coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------------
Intercept                               -0.1431      0.3

**RT10**

resolved_transfer ~ Focus * Agent + Goal_Type

In [155]:
RT10 = smf.logit(
    "resolved_transfer ~ Focus * Agent + Goal_Type",
    data=goals_escapees
    ).fit()

print(RT10.summary())

Optimization terminated successfully.
         Current function value: 0.634611
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                  106
Model:                          Logit   Df Residuals:                      101
Method:                           MLE   Df Model:                            4
Date:                Thu, 23 Jul 2026   Pseudo R-squ.:                 0.06490
Time:                        18:26:41   Log-Likelihood:                -67.269
converged:                       True   LL-Null:                       -71.938
Covariance Type:            nonrobust   LLR p-value:                   0.05318
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                         -0.7155      0.425     -1.682     

For the full model on goal split data, Focus and Agent keep similar significance and the interaction's coefficient, while not significant, is similar to the previous one and the same direction. 
Goal type's contribution remains very steady, with a very similar contribution (coeff = 0.6518) compared to the additive model Focus-Goal_Type RT05 (coeff = 0.6531) and p-value = 0.115 (vs. 0.108)

**No Goals**

**Main Effect Models**

**RT11**

resolved_transfer ~ Focus

In [156]:
RT11= smf.logit(
    "resolved_transfer ~ Focus",
    data=no_goal_escapees
    ).fit()

print(RT11.summary())

Optimization terminated successfully.
         Current function value: 0.659148
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   54
Model:                          Logit   Df Residuals:                       52
Method:                           MLE   Df Model:                            1
Date:                Thu, 23 Jul 2026   Pseudo R-squ.:               6.712e-06
Time:                        18:26:41   Log-Likelihood:                -35.594
converged:                       True   LL-Null:                       -35.594
Covariance Type:            nonrobust   LLR p-value:                    0.9826
                    coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept         0.5261      0.350      1.504      0.133      -0.160       1.212
Focus[T.They]     0.

**RT12**

resolved_transfer ~ Agent

In [157]:
RT12= smf.logit(
    "resolved_transfer ~ Agent",
    data=no_goal_escapees
    ).fit()

print(RT12.summary())

Optimization terminated successfully.
         Current function value: 0.651219
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   54
Model:                          Logit   Df Residuals:                       52
Method:                           MLE   Df Model:                            1
Date:                Thu, 23 Jul 2026   Pseudo R-squ.:                 0.01204
Time:                        18:26:41   Log-Likelihood:                -35.166
converged:                       True   LL-Null:                       -35.594
Covariance Type:            nonrobust   LLR p-value:                    0.3546
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.7376      0.367      2.012      0.044       0.019       1.456
Agent         -0.5369      0.

**Additive Model**

**RT13**

resolved_transfer ~ Focus + Agent

In [158]:
RT13= smf.logit(
    "resolved_transfer ~ Focus + Agent",
    data=no_goal_escapees
    ).fit()

print(RT13.summary())

Optimization terminated successfully.
         Current function value: 0.651190
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   54
Model:                          Logit   Df Residuals:                       51
Method:                           MLE   Df Model:                            2
Date:                Thu, 23 Jul 2026   Pseudo R-squ.:                 0.01208
Time:                        18:26:41   Log-Likelihood:                -35.164
converged:                       True   LL-Null:                       -35.594
Covariance Type:            nonrobust   LLR p-value:                    0.6505
                    coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept         0.7505      0.433      1.734      0.083      -0.098       1.599
Focus[T.They]    -0.

**Interactive Model**

**RT14**

resolved_transfer ~ Focus * Agent

In [159]:
RT14= smf.logit(
    "resolved_transfer ~ Focus * Agent",
    data=no_goal_escapees
    ).fit()

print(RT14.summary())

Optimization terminated successfully.
         Current function value: 0.628175
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:      resolved_transfer   No. Observations:                   54
Model:                          Logit   Df Residuals:                       50
Method:                           MLE   Df Model:                            3
Date:                Thu, 23 Jul 2026   Pseudo R-squ.:                 0.04700
Time:                        18:26:41   Log-Likelihood:                -33.921
converged:                       True   LL-Null:                       -35.594
Covariance Type:            nonrobust   LLR p-value:                    0.3413
                          coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------------
Intercept               0.4855      0.449      1.080      0.280      -0.395       1.366
Fo

As expected, once no_goals are analysed on their own, 54 observations are truly not enough to see anything. 

4. Model Comparisons

In [160]:
from scipy.stats import chi2

def lr_test(model_small, model_large):

    LR = 2 * (model_large.llf - model_small.llf)

    df = model_large.df_model - model_small.df_model

    p = chi2.sf(LR, df)

    return pd.Series({
        "LR": LR,
        "df": df,
        "p_value": p
    })

In [161]:
lr_test(RT02, RT03)

LR         2.368322
df         0.000000
p_value         NaN
dtype: float64

In [162]:
lr_test(RT02, RT04)

LR         3.91982
df         1.00000
p_value    0.04772
dtype: float64

5. Summary Table

In [163]:
goal_behaviour = ["Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "No_goal", "No_goal", "No_goal", "No_goal", "No_goal", "No_goal", "No_goal"]

model_name = ["RT01", "RT02", "RT03", "RT04", "RT04", "RT05", "RT05", "RT06", "RT06", "RT07", "RT07", "RT07", "RT08", "RT08", "RT08", "RT09", "RT09", "RT09", "RT10", "RT10", "RT10", "RT10", "RT11", "RT12", "RT13", "RT13", "RT14", "RT14", "RT14"]

outcome = ["resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer", "resolved_transfer"]

formula = ["resolved_transfer ~ Focus", "resolved_transfer ~ Agent", "resolved_transfer ~ Goal_Type", "resolved_transfer ~ Focus + Agent", "resolved_transfer ~ Focus + Agent", "resolved_transfer ~ Focus + Goal_Type", "resolved_transfer ~ Focus + Goal_Type", "resolved_transfer ~ Agent + Goal_Type", "resolved_transfer ~ Agent + Goal_Type", "resolved_transfer ~ Focus * Agent", "resolved_transfer ~ Focus * Agent", "resolved_transfer ~ Focus * Agent", "resolved_transfer ~ Focus * Goal_Type", "resolved_transfer ~ Focus * Goal_Type", "resolved_transfer ~ Focus * Goal_Type", "resolved_transfer ~ Agent * Goal_Type", "resolved_transfer ~ Agent * Goal_Type", "resolved_transfer ~ Agent * Goal_Type", "resolved_transfer ~ Focus * Agent + Goal_Type", "resolved_transfer ~ Focus * Agent + Goal_Type", "resolved_transfer ~ Focus * Agent + Goal_Type", "resolved_transfer ~ Focus * Agent + Goal_Type", "resolved_transfer ~ Focus", "resolved_transfer ~ Agent", "resolved_transfer ~ Focus + Agent", "resolved_transfer ~ Focus + Agent", "resolved_transfer ~ Focus * Agent", "resolved_transfer ~ Focus * Agent", "resolved_transfer ~ Focus * Agent" ]

predictor= ["They", "Agent", "goal_non_frequent", "They", "Agent", "They", "goal_non_frequent", "goal_non_frequent", "Agent", "They", "Agent", "They:Agent", "They", "goal_non_frequent", "They:goal_non_frequent", "goal_non_frequent", "Agent", "Agent:goal_non_frequent", "They", "goal_non_frequent", "Agent", "They:Agent", "They", "Agent", "They", "Agent", "They", "Agent", "They:Agent"]

coef = [0.7621, 0.2685, 0.6665, 0.8133, 0.3768, 0.7500, 0.6531, 0.6786, 0.2967, 1.3446, 0.8720, -1.3262, 0.6737, 0.5937, 0.1535, 0.7027, 0.3254, -0.0584, 1.3212, 0.6518, 0.8835, -1.2910, 0.0129, -0.5369, -0.0336, -0.5397, 0.7185, 0.1023, -1.9994]

std_error = [0.411, 0.401, 0.400, 0.418, 0.413, 0.416, 0.406, 0.401, 0.407, 0.550 , 0.525, 0.846, 0.587, 0.518, 0.834, 0.524, 0.572, 0.814, 0.556, 0.413, 0.532, 0.856, 0.590, 0.580, 0.597, 0.582, 0.797, 0.716, 1.302]

coef_p = [0.064, 0.503, 0.095, 0.052, 0.362, 0.072, 0.108, 0.091, 0.466, 0.014, 0.097, 0.117, 0.251, 0.252, 0.854, 0.180, 0.569, 0.943, 0.018, 0.115, 0.097, 0.132, 0.983, 0.355, 0.955, 0.354, 0.367, 0.886, 0.125]

LLR_p = [0.06029, 0.5024, 0.09320, 0.1125, 0.1125, 0.04623, 0.04623, 0.1871, 0.1871, 0.07794, 0.07794, 0.07794, 0.1031, 0.1031, 0.1031, 0.3398, 0.3398, 0.3398, 0.05318, 0.05318, 0.05318, 0.05318, 0.9826, 0.3546, 0.6505, 0.6505, 0.3413, 0.3413, 0.3413]

observations= [106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 54, 54, 54, 54, 54, 54, 54]

In [164]:
goal_split_transfer_summary = pd.DataFrame({
    "Goal_Behaviour": goal_behaviour,
    "Model": model_name,
    "Outcome": outcome,
    "Formula": formula, 
    "Predictor": predictor,
    "Coef": coef,
    "Std_err": std_error,
    "Coef_p": coef_p,
    "Model_LLR_p": LLR_p,
    "N": observations,
})

goal_split_transfer_summary

,Goal_Behaviour,Model,Outcome,Formula,Predictor,Coef,Std_err,Coef_p,Model_LLR_p,N
0,Goal,RT01,resolved_transfer,resolved_transfer ~ Focus,They,0.7621,0.411,0.064,0.06029,106
1,Goal,RT02,resolved_transfer,resolved_transfer ~ Agent,Agent,0.2685,0.401,0.503,0.50240,106
2,Goal,RT03,resolved_transfer,resolved_transfer ~ Goal_Type,goal_non_frequent,0.6665,0.400,0.095,0.09320,106
3,Goal,RT04,resolved_transfer,resolved_transfer ~ Focus + Agent,They,0.8133,0.418,0.052,0.11250,106
4,Goal,RT04,resolved_transfer,resolved_transfer ~ Focus + Agent,Agent,0.3768,0.413,0.362,0.11250,106
5,Goal,RT05,resolved_transfer,resolved_transfer ~ Focus + Goal_Type,They,0.7500,0.416,0.072,0.04623,106
6,Goal,RT05,resolved_transfer,resolved_transfer ~ Focus + Goal_Type,goal_non_frequent,0.6531,0.406,0.108,0.04623,106
7,Goal,RT06,resolved_transfer,resolved_transfer ~ Agent + Goal_Type,goal_non_frequent,0.6786,0.401,0.091,0.18710,106
8,Goal,RT06,resolved_transfer,resolved_transfer ~ Agent + Goal_Type,Agent,0.2967,0.407,0.466,0.18710,106
9,Goal,RT07,resolved_transfer,resolved_transfer ~ Focus * Agent,They,1.3446,0.550,0.014,0.07794,106
